core/src/test/java/io/grpc/MetadataTest.java

```diff
@@ -189,6 +193,13 @@ public void roundTripIntegerMarshaller() {
     roundTripInteger(0x87654321);
   }
 
+  @Test
+  public void invalidBinaryKeyName() {
```

Comment:
You're only testing one of the conditions here ... should probably also test something that doesn't end with "-bin".